pages/pledge/[id].page.js

```diff
@@ -94,9 +101,13 @@ const PledgePage = () => {
             <Text>Revenue: ${campaign.fiatAmount}</Text>
             <Text>Pledged: ${campaign.currentAmount[0].amount}</Text>
 
-            <Button onClick={() => setIsModalOpen((prev) => !prev)}>
-              Pledge Now
-            </Button>
+            {isOwner ? (
+              <Button onClick={() => write?.()}>Claim</Button>
```

Comment:
What's the explanation for using `?.()`?

Reply:
Here is [Wagmi's documentation](https://wagmi.sh/react/hooks/useContractWrite) on how to implement a button that executes the write function.

The reason is that write can become null if there is a problem in the connection with the contract. For example: if a requirement of the contract is that the time in which it is expected to collect funds for a campaign has been fulfilled and it has not been fulfilled, write will be null.

Reply:
I'll add the disabled prop